eng/packages/http-client-csharp/generator/TestProjects/Local/Mgmt-TypeSpec/src/Generated/RestOperations/FoosRestOperations.cs

```diff
@@ -35,322 +32,81 @@ internal Foos(HttpPipeline pipeline, Uri endpoint, string apiVersion)
-        public virtual async Task<Response> ListAsync(Guid subscriptionId, string resourceGroupName, RequestContext context)
-        {
-            Argument.AssertNotNull(resourceGroupName, nameof(resourceGroupName));
-
-            using HttpMessage message = CreateListRequest(subscriptionId, resourceGroupName, context);
-            return await Pipeline.ProcessMessageAsync(message, context).ConfigureAwait(false);
-        }
-
-        /// <summary> List Foo resources by resource group. </summary>
-        /// <param name="subscriptionId"> The ID of the target subscription. The value must be an UUID. </param>
-        /// <param name="resourceGroupName"> The name of the resource group. The name is case insensitive. </param>
-        /// <param name="cancellationToken"> The cancellation token that can be used to cancel the operation. </param>
-        /// <exception cref="ArgumentNullException"> <paramref name="resourceGroupName"/> is null. </exception>
-        /// <exception cref="RequestFailedException"> Service returned a non-success status code. </exception>
-        public virtual Response<FooListResult> List(Guid subscriptionId, string resourceGroupName, CancellationToken cancellationToken = default)
-        {
-            Argument.AssertNotNull(resourceGroupName, nameof(resourceGroupName));
-
-            Response result = List(subscriptionId, resourceGroupName, cancellationToken.CanBeCanceled ? new RequestContext { CancellationToken = cancellationToken } : null);
-            return Response.FromValue((FooListResult)result, result);
-        }
-
-        /// <summary> List Foo resources by resource group. </summary>
-        /// <param name="subscriptionId"> The ID of the target subscription. The value must be an UUID. </param>
-        /// <param name="resourceGroupName"> The name of the resource group. The name is case insensitive. </param>
-        /// <param name="cancellationToken"> The cancellation token that can be used to cancel the operation. </param>
-        /// <exception cref="ArgumentNullException"> <paramref name="resourceGroupName"/> is null. </exception>
-        /// <exception cref="RequestFailedException"> Service returned a non-success status code. </exception>
-        public virtual async Task<Response<FooListResult>> ListAsync(Guid subscriptionId, string resourceGroupName, CancellationToken cancellationToken = default)
-        {
-            Argument.AssertNotNull(resourceGroupName, nameof(resourceGroupName));
-
-            Response result = await ListAsync(subscriptionId, resourceGroupName, cancellationToken.CanBeCanceled ? new RequestContext { CancellationToken = cancellationToken } : null).ConfigureAwait(false);
-            return Response.FromValue((FooListResult)result, result);
+        internal HttpMessage CreateCreateOrUpdateRequest(Guid subscriptionId, string resourceGroupName, string fooName, RequestContent content, RequestContext context)
```

Comment:
so we made a step ahead that in the rest client, we no longer keep the protocol methods and convenience methods, and the resource class would just call the create request methods here.

Do we consider making another step ahead that we could remove this class, and move all these create request methods into the resource class, and put them in another partial class like `FooResource.RestClient.cs`?
This could save us an object in resource class - now it has to construct this class to create the request.

Reply:
Currently, I have removed `Foos.RestClient.cs` and put its `CreateRequest` methods to `FoosRestOperations.cs`. I am fine with renaming it to `Foos.RestClient.cs`. The thing is, we only keep the internal `CreateRequest` methods.